### PR TITLE
Clean network

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/include/net/l3/icmpv6.h [hunk 1]
```diff
@@ -134,10 +134,10 @@ typedef struct icmp6hdr {
 		struct ndpbody_router_advert router_advert;
 		struct ndpbody_neighbor_solicit neighbor_solicit;
 		struct ndpbody_neighbor_advert neighbor_advert;
-	} __attribute__((packed)) body[];
+	} __attribute__((packed)) body;
 } __attribute__((packed)) icmp6hdr_t;
 
-#define ICMP6_MIN_HEADER_SIZE (sizeof(struct icmp6hdr))
+#define ICMP6_MIN_HEADER_SIZE (4)
```
Comment (Deryugin) [inline src/include/net/l3/icmpv6.h]: I think it's a good idea to write comment why it's 4 (`sizeof(struct icmp6hdr)` without body).

File: src/include/net/mii.h [hunk 2]
```diff
@@ -121,6 +129,38 @@
...
+ * currently supported media, in the priority order defined by
+ * IEEE 802.3u.  We use LPA_xxx constants but note this is not the
+ * value of LPA solely, as described above.
+ *
+ * The one exception to IEEE 802.3u is that 100baseT4 is placed
+ * between 100T-full and 100T-half.  If your phy does not support
+ * 100T4 this is fine.  If your phy places 100T4 elsewhere in the
+ * priority order, you will need to roll your own function.
+ */
+static inline unsigned int mii_nway_result (unsigned int negotiated)
```
Comment (Deryugin) [inline src/include/net/mii.h]: `mii_nway_result (...)` -> `mii_nway_result(...) {` ->

File: src/net/skbuff/skb_data.c [hunk 3]
```diff
@@ -57,17 +57,12 @@ struct sk_buff_data_fixed {
...
-
-	char __data[];
-} DATA_ATTR;
-
 POOL_DEF(skb_data_pool, struct sk_buff_data_fixed, MODOPS_AMOUNT_SKB_DATA);
 
-void *skb_get_data_pointner(struct sk_buff_data *skb_data) {
-	return skb_data->__data + IP_ALIGN_SIZE;
+void *skb_get_data_pointner(struct sk_buff_data *data) {
+	struct sk_buff_data_fixed *skb_data = (void*)data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void*)`->`(void *)`

File: src/net/skbuff/skb_data.c [hunk 4]
```diff
@@ -113,10 +108,11 @@ struct sk_buff_data * skb_data_alloc(size_t size) {
 	skb_data->alloc_type = alloc_type;
 	skb_data->links = 1;
 
-	return skb_data;
+	return (void *)skb_data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void *)s...`->`(void *) s...`

File: src/net/skbuff/skb_data.c [hunk 5]
```diff
@@ -113,10 +108,11 @@ struct sk_buff_data * skb_data_alloc(size_t size) {
...
 }
 
-struct sk_buff_data * skb_data_clone(struct sk_buff_data *skb_data) {
+struct sk_buff_data * skb_data_clone(struct sk_buff_data *data) {
+	struct sk_buff_data_fixed *skb_data = (void*)data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void*)data;`->`(void *) data;`

File: src/net/skbuff/skb_data.c [hunk 6]
```diff
@@ -127,14 +123,16 @@ struct sk_buff_data * skb_data_clone(struct sk_buff_data *skb_data) {
 	}
 	ipl_restore(sp);
 
-	return skb_data;
+	return (void *)skb_data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void *)skb_data;`->`(void *) skb_data;

File: src/net/skbuff/skb_data.c [hunk 7]
```diff
@@ -127,14 +123,16 @@ struct sk_buff_data * skb_data_clone(struct sk_buff_data *skb_data) {
...
 }
 
-int skb_data_cloned(const struct sk_buff_data *skb_data) {
+int skb_data_cloned(const struct sk_buff_data *data) {
+	struct sk_buff_data_fixed *skb_data = (void*)data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void*)data;`->`(void *) data;`

File: src/net/skbuff/skb_data.c [hunk 8]
```diff
@@ -127,14 +123,16 @@ struct sk_buff_data * skb_data_clone(struct sk_buff_data *skb_data) {
...
 	return skb_data->links != 1;
 }
 
-void skb_data_free(struct sk_buff_data *skb_data) {
+void skb_data_free(struct sk_buff_data *data) {
+	struct sk_buff_data_fixed *skb_data = (void*)data;
```
Comment (Deryugin) [inline src/net/skbuff/skb_data.c]: `(void*)data;`->`(void *) data;`

File: src/include/net/l3/icmpv6.h (hunk 1 above)
Comment (anton-bondarev) [reply src/include/net/l3/icmpv6.h]: thanks. done

File: src/include/net/mii.h (hunk 2 above)
Comment (anton-bondarev) [reply src/include/net/mii.h]: done

File: src/net/skbuff/skb_data.c (hunk 3 above)
Comment (anton-bondarev) [reply src/net/skbuff/skb_data.c]: done

File: src/net/skbuff/skb_data.c (hunk 4 above)
Comment (anton-bondarev) [reply src/net/skbuff/skb_data.c]: In my mind
`(cast type)var` looks better than '(cast type) var'

**Pull request merged**